src/backend/egl/display.rs

```diff
@@ -799,7 +799,17 @@ fn get_dmabuf_formats(
 
     for fourcc in formats {
         let mut num = 0i32;
-        wrap_egl_call(|| unsafe {
+        // Some drivers return EGL_BAD_PARAMETER here for formats
+        // they themselves returned for the query above.. *sigh*
+        //
+        // - NVIDIA proprietary since 520
+        //
+        // So lets ignore any errors of this call on purpose(!),
+        // which will let `num` stay at `0` and handle the format
+        // as unsupported by explicit modifiers.
+        // Which is probably what the error is suppose to indicate
+        // although the spec doesn't seem to demand it...
```

Comment:
This is indeed a driver bug as it disagrees with the spec.

I checked the EGL CTS and there is only testing for whether eglGetProcAddress exposed the symbols and not whether the implementation behaves properly: https://github.com/KhronosGroup/VK-GL-CTS/search?p=2&q=EGL_EXT_image_dma_buf_import_modifiers&type=

> * If <format> is not one of the formats advertised by QueryDmaBufFormatsEXT for the same <dpy>, the error EGL_BAD_PARAMETER is generated.

Have you filed a driver bug with Nvidia @Drakulix?

I guess if we wanted to be extra sure, one of us could write a pull request for the EGL CTS and add a new test for this?